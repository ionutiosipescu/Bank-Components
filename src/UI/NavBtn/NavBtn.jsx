import React, { useState } from "react";
import { Link } from "react-router-dom";
import Button from "../Button/Button";

import { Label, NavBtnContainer } from "./NavBtn.style";
import { PropTypes } from "prop-types";
import LinkButton from "../LinkButton/LinkButton";

function NavBtn({ to, children, label, active, selected, onClick, size }) {
  return (
    <NavBtnContainer selected={selected} onClick={onClick}>
      <LinkButton to={to} size={size}>
        {children}
      </LinkButton>
      <Label as={Link} to={to} active={active}>
        {label}
      </Label>
    </NavBtnContainer>
  );
}

NavBtn.propTypes = {
  to: PropTypes.string,
  label: PropTypes.string,
  children: PropTypes.element,
};

export default NavBtn;
