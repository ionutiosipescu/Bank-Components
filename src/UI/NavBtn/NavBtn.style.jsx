import styled from "styled-components";

export const NavBtnContainer = styled.div`
  display: flex;
  align-items: center;
  width: 80%;
  cursor: pointer;
  ${(props) =>
    props.selected === true
      ? `& :nth-child(1) {
    background-color: var(--purple);
    color: var(--text-light);
    transition: 0.3s;
  }
  & * {
    opacity: 1;
  }`
      : ""}
`;

export const Label = styled.h2`
  font-weight: 600;
  opacity: 0.7;
  color: var(--text);
  &:hover {
    opacity: 1;
  }
  ${(props) =>
    props.active === "active"
      ? "display: block; margin-left: 20px;"
      : "display: none;"}
`;
