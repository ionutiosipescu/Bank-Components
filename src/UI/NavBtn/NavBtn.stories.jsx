import { BsHouseDoor } from "react-icons/bs";
import NavBtn from "./NavBtn";

export default {
  title: "Components/NavBtn",
  component: NavBtn,
};

const Template = (args) => <NavBtn {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  label: "Home",
  children: <BsHouseDoor size={22} />,
};
