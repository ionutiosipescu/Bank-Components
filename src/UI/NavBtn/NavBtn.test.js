import React from "react";
import { shallow, mount, render } from "enzyme";
import NavBtn from "./NavBtn";

it("expect to render NavBtn component", () => {
  expect(shallow(<NavBtn />)).toMatchSnapshot();
});
