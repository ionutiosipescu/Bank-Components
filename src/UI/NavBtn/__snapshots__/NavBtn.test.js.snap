// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render NavBtn component 1`] = `
<styled.div>
  <LinkButton />
  <styled.h2
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
  />
</styled.div>
`;
